contracts/src/NStates.sol

```diff
@@ -80,46 +80,58 @@ contract NStates is IncrementalMerkleTree {
      */
     function move(
         uint256[7] memory pubSignals,
-        uint256[2] memory a,
-        uint256[2][2] memory b,
-        uint256[2] memory c
+        uint256[8] memory formattedProof,
+        uint8 v,
+        bytes32 r,
+        bytes32 s
     ) public {
-        uint256 root = pubSignals[0];
-        uint256 troopInterval = pubSignals[1];
-        uint256 waterInterval = pubSignals[2];
-        uint256 hUFrom = pubSignals[3];
-        uint256 hUTo = pubSignals[4];
-        uint256 rhoFrom = pubSignals[5];
-        uint256 rhoTo = pubSignals[6];
-
-        require(rootHistory[root], "Root must be in root history");
+        // root = pubSignals[0];
+        // troopInterval = pubSignals[1];
+        // waterInterval = pubSignals[2];
+        // hUFrom = pubSignals[3];
+        // hUTo = pubSignals[4];
+        // rhoFrom = pubSignals[5];
+        // rhoTo = pubSignals[6];
+        require(rootHistory[pubSignals[0]], "Root must be in root history");
         require(
-            currentTroopInterval() >= troopInterval,
+            currentTroopInterval() >= pubSignals[1],
             "Move is too far into the future, change currentTroopInterval value"
         );
         require(
-            currentWaterInterval() >= waterInterval,
+            currentWaterInterval() >= pubSignals[2],
             "Move is too far into the future, change currentWaterInterval value"
         );
         require(
-            !nullifiers[rhoFrom] && !nullifiers[rhoTo],
+            !nullifiers[pubSignals[5]] && !nullifiers[pubSignals[6]],
             "Move has already been made"
         );
         require(
-            verifierContract.verifyProof(a, b, c, pubSignals),
+            getSigner(pubSignals[3], pubSignals[4], v, r, s) == owner,
```

Comment:
could you make this a contract deploy env variable rather than msg.owner 

you can hardcode it to our dev privkey so it'll be effectively the same. want to clearly have two separate entities here: 1) the enclave with its built-in keypair and 2) the devs deploying the contract

Reply:
it'll be effectively the same for now bc we'll use the dev keypair for the enclave signing (it would be annoying to have to redeploy the contract whenever the enclave restarts)

good to properly structure it now though so we're ready when we go live & the enclave randomly generates the signing keypair during start up

Reply:
"contract deploy env variable" up there btw means adding to `forge_create_local.sh`

Reply:
This is a good idea. But we won't know the enclave's address before deploying the contract right?

Reply:
is this for prod or dev? when we go to prod, we'll have the enclave running first and deploy the contract after

for dev, that's definitely true, so we can hardcode